### PR TITLE
Fixes to Validate King

### PR DESCRIPTION
**Pull request by huntedhippo**
Castling if statement rejects king moves that don't move to file 2 or file 6.  This prevents the king from making a normal move on his first move.  Therefore added 'absolute_file_delta > 1' as if condition so that only castling moves are caught by the castling if statement and normal moves are handled below it.

Removed hypothetical_check of final position as this is done in validate_move.  Added hypothetical check of current position and position where the rook ends (and the king passes through).

Added test that king does not move into his own piece.

### CODE REVIEW

File: server/gamemodes/chess.py [hunk 1]
```diff
@@ -256,7 +256,9 @@ def validate_king_move(
         """Validate a king's move."""
         absolute_file_delta = abs(file - king.file)
         absolute_rank_delta = abs(rank - king.rank)
-        if (not absolute_rank_delta) and not king.has_moved:
+        #if statement to handle where castling is attempted
+        if (not absolute_rank_delta) and not king.has_moved and absolute_file_delta > 1:
+            #check that castling is valid
```
Comment (Artemis21) [inline server/gamemodes/chess.py]:
```suggestion
        # Check for castling attempt.
        if (not absolute_rank_delta) and not king.has_moved and absolute_file_delta > 1:
```
Comments should describe what code does, not how it does it - we can see it's an if statement.

File: server/gamemodes/chess.py [hunk 2]
```diff
@@ -271,14 +273,19 @@ def validate_king_move(
...
             for empty_file in empty_files:
-                if not self.get_piece(rank, empty_file):
+                if self.get_piece(rank, empty_file):
                     return False
-            return not self.hypothetical_check(
-                king.side, (king, rank, file), (rook, rank, rook_end)
-            )
+            if self.hypothetical_check(king.side):
+                return False
+            if self.hypothetical_check(king.side,(king,rank,rook_end)):
```
Comment (Artemis21) [inline server/gamemodes/chess.py]:
```suggestion
            if self.hypothetical_check(king.side, (king, rank, rook_end)):
```
Lint!

**Review by Artemis21 — CHANGES_REQUESTED** [review]
Looks good, a couple of improvements to style.

File: server/gamemodes/chess.py (hunk 2 above)
Comment (huntedhippo) [reply server/gamemodes/chess.py]: Sorry about this, I thought Visual Studio 2019 was linting for me.  I may need to move to VS Code as I can't seem to figure out how to lint with VS2019.

**Review by Artemis21 — CHANGES_REQUESTED** [review]
Please remove .vs/slnx.sqlite

**Comment by huntedhippo** [comment]
Now done.  Annoying that VS19 wouldn't let me delete this.

**Review by Artemis21 — APPROVED** [review]

**Comment by Artemis21** [comment]
9 down, 1 to go!

**Pull request merged**